### DISCUSSION TITLE
Confused on how to handle schedule and triggers with Job

### DISCUSSION THREAD

**Discussion by CrazyMoka** [quartznet/quartznet#1648] [Q&A]
Hello,
Wondering how I should handle triggers, Job and schedule.
Say I have N jobs, one can run on demand and it also needs to be run at Midnight.  Do I create a schedule for the midnight job with one trigger?
What happens if I have other jobs I want to run at midnight do I create new triggers with midnight schedule or can I share the same schedule for multiple jobs via triggers?

Also note I need to keep track of things running in a basic admin dashboard so I am using Database to store information.

Thanks

**Comment by CrazyMoka** [comment]
After reading and messing arround.
I guess the real question is, how can I attach one job to many triggers?
                ITrigger trigger2 = await scheduler.GetTrigger(new TriggerKey("myTrigger", "group1"));
                IJobDetail job3 = await scheduler.GetJobDetail(jobKey); 
                await scheduler.ScheduleJob(job3,trigger2);
I get this error
SchedulerException: Trigger does not reference given job!

The job already exists and I'm retreiving it and I want to add it to a trigger that also already exists.

**Reply by ghost** [reply]
That is because ScheduleJob will ADD and SAVE the JobDetail when you supply it as a parameter. Try getting the TriggerBuilder, then specify what job it is for, and then omit the job parameter this:

            ITrigger trigger2 = await s.GetTrigger(new TriggerKey("myTrigger", "group1"));
            trigger2.GetTriggerBuilder().ForJob(jobKey);
            IJobDetail job3 = await s.GetJobDetail(jobKey);
            await s.ScheduleJob(trigger2);

Note, I have tested this code... and there might be a better way to do it.

**Reply by CrazyMoka** [reply]
I also trying that too but it wont work saying the trigger already exists.
ObjectAlreadyExistsException: Unable to store Trigger: 'group1.myTrigger', because one already exists with this identification.
So I can't use the same trigger for multitple jobs?

**Reply by CrazyMoka** [reply]
Right from the docs:
"
For example, Jobs can be created and stored in the job scheduler independent of a trigger, and many triggers can be associated with the same job. Another benefit of this loose-coupling is the ability to configure jobs that remain in the scheduler after their associated triggers have expired, so that that it can be rescheduled later, without having to re-define it. It also allows you to modify or replace a trigger without having to re-define its associated job.
"

Apparently not if I can't add a job to a trigger and I'm getting a trigger exists error.
Any examples to show how to add multiple jobs to the same trigger,
After retreiving an already made trigger?

**Reply by CrazyMoka** [reply]
I feel like an idiot.  I was reading it as one job can belong to multiple triggers. Where in fact its the other way arround. One trigger belong to multiple jobs .... :( 
I'm such a goof.

My question still remains.... Why would I want to create multiple triggers to run a midnight for each job when I could make one trigger to run multiple jobs at midnight?   

So I have to make 15 triggers that run at midnight if I have 15 jobs that need to be run at midnight??
